websites/V/Vostfree/presence.ts

```diff
@@ -0,0 +1,102 @@
+const presence = new Presence({
+    clientId: "925799512205844490"
+  }),
+  path = document.location.pathname,
+  browsingStamp = Math.floor(Date.now() / 1000),
+  search = document.querySelector("#dle-content > h2"),
+  getGenreTitle = (genre: string) => {
+    [, , genre] = genre.split("/");
+    return genre;
+  };
+
+let video = {
+  duration: 0,
+  currentTime: 0,
+  paused: true
+};
+
+presence.on(
+  "iFrameData",
+  (data: { duration: number; currentTime: number; paused: boolean }) => {
+    if (data?.duration ? true : false) video = data;
+  }
+);
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+      largeImageKey: "logo",
+      startTimestamp: browsingStamp
+    },
+    [privacy, button] = await Promise.all([
+      presence.getSetting<boolean>("privacy"),
+      presence.getSetting<boolean>("button")
+    ]),
+    title = document.querySelector(
+      "#dle-content > div.watch-top > div > div > div > div.slide-middle > h1"
+    );
+
+  if (!isNaN(video.duration) && title) {
+    const episode = document.querySelector(
+        "#player-tabs > div.tab-blocks > div:nth-child(1) > div > div.new_player_top > div.new_player_selector_box > div.jq-selectbox-wrapper > div > div.jq-selectbox__select > div.jq-selectbox__select-text"
+      ),
+      [startTimestamp, endTimestamp] = presence.getTimestamps(
+        video.currentTime,
+        video.duration
+      );
+    presenceData.details = title.textContent;
+    presenceData.state = episode?.textContent;
+    presenceData.startTimestamp = startTimestamp;
+    presenceData.endTimestamp = endTimestamp;
+    presenceData.buttons = [
+      {
+        label: "Regarder l'épisode",
+        url: document.location.href
+      }
+    ];
+    presenceData.smallImageKey = video.paused ? "pause" : "play";
+    presenceData.smallImageText = video.paused ? "En pause" : "En cours";
+
+    if (!button) delete presenceData.buttons;
+
+    if (privacy) {
+      delete presenceData.state;
+      delete presenceData.endTimestamp;
+      delete presenceData.buttons;
+      presenceData.details = "Regarde un anime";
+      presenceData.startTimestamp = browsingStamp;
```

Comment:
Already set in the obj, uneeded
```suggestion
```

Reply:
if I'm removing it, it will reset the timestamp everytime I pause it